imap_processing/utils.py

```diff
@@ -254,3 +255,86 @@ def update_epoch_to_datetime(dataset: xr.Dataset) -> xr.Dataset:
     dataset = dataset.assign_coords(epoch=epoch)
 
     return dataset
+
+
+def packet_file_to_datasets(
```

Comment:
I see quite a bit of overlap with the `create_dataset` function. Some differences are:

- This function creates a dataset for each apid in the input file
- `create_dataset` has an option for including the packet header or not
- `create_dataset` brings in variable attributes from a source external to the xtce. Right now that source has not been updated to use the ImapCdfAttributes class.
- `create_dataset` allows for the option to skip certain fields in the packet.

Reply:
I agree, this is heavily based on that function. Would you want all of these capabilities? My thought is that sometimes too many options is not great either and sometimes standardizing on a fixed layout is useful.

I'd actually rather not "skip" the specific fields and complicate this function, but rather just have a user do `dataset.drop("variable")` to remove whatever they want afterwards. So I kind of view this as wanting to give the user everything and then the user can start winnowing down to what they need or transforming from there.

If it would be useful to add some ignore keywords I am open to it though.

Reply:
I was really just making an observation about the overlap. I am in agreement that keeping this function simple makes sense.

Reply:
I agree. I think it's gonna save lot of repeated steps that we all are doing separately. This function seems much better. It's decomming, creating dataset and then returning it per apid. Then every processing that happens after that can decide what to keep and what to drop in terms of the actual data. And it's already in the format we need to write to cdf except we need to do minor things to add attrs stuff.